pure2.md

```diff
@@ -0,0 +1,535 @@
+
+4.  Device sends a new signed manifest indicating the success (or failure) of the update.
+
+In principle, this process could be adapted for offline use cases by pre-generating Director metadata for each potential device and distributing this metadata via a side channel. However, this is quite difficult in practical application, and doesn't address any of the use cases involving updates to devices whose ECU keys and serial numbers are not yet registered with the inventory database, such as factory pre-provisioning.
+
+There are a number of potential scenarios where an OEM may wish to validate and secure updates that are difficult to address with the full Uptane protocol. This is especially true in non-automotive uses of Uptane, such as in industrial automation systems or medical devices. Industrial automation systems may be on completely airgapped networks, never able to connect to update servers. Medical devices may be connected to the internet only very rarely, but still desire the advanced security benefits (in particular, selectively directed updates) that Uptane can provide over The Update Framework (TUF). Specific use cases for this PURE include:
+
+-   Initial load-out of software on the factory floor, before all ECUs are present and before they are registered with the inventory database
+
+-   Updating devices on airgapped networks where a full Uptane implementation may not be practical
+
+-   Allowing service technicians to deliver updates at their discretion, but only from an OEM-directed set of possible updates
+
+## Specification
+
+### The offline-update metadata role 
+
+To solve the problems above, we add a new role to the Director repository, called the "offline-update" role. The new offline-update role is responsible for producing and signing metadata about sets of images that are valid for offline updates. This includes two types of metadata: offline-update snapshot metadata and offline-update targets metadata. These roles are analagous to snapshot and targets metadata on the image repository.
+
+Each offline-update targets metadata file lists a set of images eligible for installation together as an offline update. This metadata is designed to be loaded onto a portable storage medium, such as a USB drive or a diagnostics/flashing tool with attached storage, alongside an up-to-date version of all non-device-specific metadata from both Uptane repositories and binaries for all targets listed. Uptane clients can check for the presence of offline-update metadata, either by an automated process (e.g. periodically checking some well-known location on a filesystem or TFTP server) or via a manually-triggered diagnostic command. If offline-update metadata is found, the clients validate it, check it against the Image repository metadata, and then install the targets.
+
+When a devices receives online updates from the Director receive the latest version of the offline-update snapshot metadata as part of the normal update cycle.
```

Comment:
This sentence does not make sense to me.

Reply:
Oops, let me fix that. I missed a word and a comma there; it should say "When a devices receives online updates from the Director, it receives the latest version of the offline-update snapshot metadata as part of the normal update cycle."

Reply:
"When a device"

Reply:
🤦

Reply:
fixed